arch/arm/core/aarch32/cortex_m/mpu/arm_core_mpu.c

```diff
@@ -223,12 +242,19 @@ void z_arm_configure_dynamic_mpu_regions(struct k_thread *thread)
 
 #if defined(CONFIG_USERSPACE)
 	if (thread->arch.priv_stack_start) {
+		/* A K_USER thread has the stack guard protecting the privilege
+		 * stack and not on the usermode stack because the user mode
+		 * stack is already has its own defined memory region.
+		 */
 		guard_start = thread->arch.priv_stack_start - guard_size;
 
 		__ASSERT((uint32_t)&z_priv_stacks_ram_start <= guard_start,
-		"Guard start: (0x%x) below privilege stacks boundary: (0x%x)",
+		"Guard start: (0x%thread->stack_info.startx) below privilege stacks boundary: (0x%x)",
```

Comment:
This is probably not intended; revert?

Reply:
yeah... sorry

Reply:
yep... clearly didn't compile before pushing.